Mathlib/RingTheory/LocalProperties/Submodule.lean

```diff
@@ -99,3 +106,182 @@ theorem Module.subsingleton_of_localization_maximal
   rw [subsingleton_iff_forall_eq 0]
   intro x
   exact Module.eq_of_localization_maximal Rₚ Mₚ f x 0 fun _ _ ↦ Subsingleton.elim _ _
+
+section localization_maximal
+
+open Submodule LocalizedModule Ideal LinearMap
+
+/-- Another version, using `LocalizedModule` instead of `IsLocalizedModule`. -/
+lemma Submodule.le_of_localization_maximal' {N H : Submodule R M}
+    (h : ∀ (J : Ideal R) [J.IsMaximal], localized J.primeCompl N ≤ localized J.primeCompl H) :
+    N ≤ H := Submodule.le_of_localization_maximal (fun P ↦ (Localization P.primeCompl))
```

Comment:
I'm worried about the inconsistency of names: for `maximal` lemmas you add a `'` for the LocalizedModule version, but for the `span` lemmas you add a `'` for the IsLocalizedModule version. I think the latter convention is better, because the LocalizedModule versions are easier to apply and should be considered the default, so their names shouldn't have a `'`.

Reply:
Oh now seeing this. I would argue the other way round or even propose, as above, `of_isLocalized` vs `of_localized`. This is clear and avoids a `'`.

Reply:
#19118 is now merged, and I'm following this `localized` vs `isLocalized` convention [in #19126](https://github.com/leanprover-community/mathlib4/pull/19126/commits/b9bce1857ba2ceb3d94215cde96679db7f772cc0), but I only applied it to new `span` lemmas but not yet the existing `maximal` lemmas, and I only introduced the `isLocalized` version. If @su00000 and @erdOne agree, let's make the `maximal` lemmas follow the new convention too, and deprecate the old names.

Reply:
Sorry for my late answer, and thanks for doing these! I'll merge #19126 to #19080. I agree.

Reply:
I've updated #19126 to implement all my suggestions; again, please feel free to merge. I think this is ready to go!